### DISCUSSION TITLE
aws credentials issue : when deploying considering only default profile name not other name (python starter template)

### DISCUSSION THREAD

**Discussion by RohitSaini15** [serverless/serverless#11552] [Q&A]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

when you create starter python template. in the **serverless.yaml** file , in the **provider** key there is no **profile** key so it is not considering any other profile key name in **aws credential file** so when you deploy it shows an error that **no credential file found**.

**Error**

sls deploy

× Stack aws-python-project2-dev failed to deploy (2s)
Environment: win32, node 16.17.0, framework 3.25.0, plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
AWS provider credentials not found. Learn how to set up AWS provider credentials in our docs here: <http://slss.io/aws-creds-setup>.

### Service configuration (serverless.yml) content

```yaml
service: aws-python-project2

frameworkVersion: '3'

provider:
  name: aws
  runtime: python3.8

functions:
  hello:
    handler: handler.hello
```

### Command name and used flags

N/A

### Command output

```shell
N/A
```

### Environment information

```shell
Framework Core: 3.25.0
Plugin: 6.2.2
SDK: 4.3.2
```

**Comment by medikoo** [comment]
Serverless Framework never did and never will pre-decide which eventually configured AWS profile should be used for authentication.

Take a look at how AWS SDK CLI works, it'll throw with same error if you won't configure AWS credentials in your environment.

**Comment by RohitSaini15** [comment]
> Serverless Framework never did and never will pre-decide which eventually configured AWS profile should be used for authentication.
> 
> Take a look at how AWS SDK CLI works, it'll throw with same error if you won't configure AWS credentials in your environment.

i have configured my aws-profile with "serverless-admin" name but still was not working

**Comment by RohitSaini15** [comment]
so i manually type profile name in the serverless.yaml then it started working

**Comment by medikoo** [comment]
@RohitSaini15 I moved this to discussion, as it's more an usage issue and not a Framework bug. You may try posting on Stack Overflow there's bigger audience ther
